ui/src/components/admin/forms/AdminFormDataViewer.js

```diff
@@ -0,0 +1,265 @@
+import React, { useEffect, useState, useRef } from 'react';
+import {} from '@chakra-ui/react';
+import { customAxios as axios } from '../../helpers/customAxios';
+import { AgGridReact } from 'ag-grid-react';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
+import { Button, Heading, Text, Input, HStack, Box } from '@chakra-ui/react';
+
+export default function AdminFormDataViewer(props) {
+  const {
+    location: { state },
+  } = props;
+  const formName = state.name;
+  const formId = state.id;
+  const [gridApi, setGridApi] = useState();
+
+  const [formData, setFormData] = useState([]);
+  const [api, setApi] = useState();
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+
+  const getFilterType = () => {
+    if (startDate !== '' && endDate !== '') return 'inRange';
+    else if (startDate !== '') return 'greaterThan';
+    else if (endDate !== '') return 'lessThan';
+  };
+
+  const exportHandler = () => {
+    if (api) {
+      api.exportDataAsCsv(exportParams());
+    }
+  };
+  const exportParams = () => {
+    var currentdate = new Date();
+    var datetime =
+      currentdate.getFullYear() +
+      '' +
+      (currentdate.getMonth() + 1) +
+      '' +
+      currentdate.getDate() +
+      '_' +
+      currentdate.getHours() +
+      '-' +
+      currentdate.getMinutes();
+    let fileName = `${formName} [${datetime}].csv`;
+
+    return {
+      skipColumnGroupHeaders: true,
+      allColumns: true,
+      fileName: fileName,
+    };
+  };
+
+  const dateFilterParams = {
+    comparator: function (filterLocalDateAtMidnight, cellValue) {
+      var dateAsString = cellValue;
```

Comment:
using luxon you probably have to convert filterLocalDateAtMidnight to luxon Datetime
```
let dateAtMidnight = DateTime.fromJSDate(filterLocalDateAtMidnight);
let dateObj = DateTime.fromFormat('yyyy-MM-dd HH:mm:ss');

if (dateAtMidnight > dateObj)
// etc.
```